packages/pagination/__tests__/pagination.spec.ts

```diff
@@ -1,407 +1,377 @@
-import { mount } from '@vue/test-utils'
-import { sleep } from '@element-plus/test-utils'
+import { mount, VueWrapper } from '@vue/test-utils'
+import { triggerEvent } from '@element-plus/test-utils'
 import Pagination from '../src/index'
-import { nextTick, ref, h } from 'vue'
+import { nextTick, ref, h, defineComponent } from 'vue'
 
 const TIME_OUT = 100
 
-describe('Pagination.vue', () => {
-  test('layout', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        layout: 'prev, pager, next',
-      },
+const assertElementsExistence = (wrapper: VueWrapper<any>, selectors: string[], existence: boolean) => {
+  selectors.forEach(selector => {
+    expect(wrapper.find(selector).exists()).toBe(existence)
+  })
+}
+
+const assertCurrent = (wrapper, page) => {
+  expect(wrapper.find('.el-pager li.active.number').text()).toBe(String(page))
+}
+const assertTotal = (wrapper, total) => {
+  expect(wrapper.find('.el-pagination .el-pager li:last-child').text()).toBe(String(total))
+}
+
+describe('Pagination', () => {
+  describe('test invalid usages', () => {
+    let warned = false
+    let cacheWarn = console.warn
+    beforeEach(() => {
+      warned = false
+      cacheWarn = console.warn
+      console.warn = () => warned = true
```

Comment:
Can we use `jest.spyOn` here? Refer to: [Jest spyOn built-in methods](https://jestjs.io/zh-Hans/docs/jest-object)

Reply:
As a replacement, I changed to `jest.fn` to hijack origin `console.warn`, because `spyOn` will cause origin warn really executed and  many warn messages displayed among testing result, that's very uply. Refer to my added commit.